src/routing_table/xorable.rs

```diff
@@ -128,6 +162,22 @@ macro_rules! impl_xorable {
                 let pow_i = 1 << (mem::size_of::<Self>() * 8 - 1 - i); // 1 on bit i.
                 self ^ pow_i
             }
+
+            fn binary(&self) -> String {
+                let bits = mem::size_of::<Self>() * 8;
+                let binary = format!("{:1b}", self);
```

Comment:
first, I don't see the padded zero being printed out, I only see they have been aligned.
second, the padded of zero will only happen when the value is less than the upper limit.
if we try to print a u64 with {:08b}, there won't be padded zeros.